utils/files.ts

```diff
@@ -0,0 +1,46 @@
+import { RESPONSE_MESSAGES } from "../constants/index";
+import { NextApiResponse } from "next";
+import { Transform } from "stream";
+
+export function resolveHeaders(headers:string[]): string {
+    return headers.join(',');
+}
+
+export function getDataFromValues(data:any, headers: string[]): any {
+    const result: any = {};
+    headers.forEach(header => {
+        result[header] = data[header];
+    });
+
+    return result;
+}
+
+export const getTransform = (headers:string[]) => new Transform({
+    objectMode: true,
+    transform(chunk: any, encoding: BufferEncoding, callback: () => void) {
+        const csvLine = headers.map(header => chunk[header]).join(',') + '\n';
```

Comment:
instead of repeating the join here you should use the function you defined above.

Reply:
Thanks for the suggestion about naming, I improved that, and also added the delimiter option and some more tests.